OneCommunityGlobal/HighestGoodNetworkApp
# Luis summary intro btn and add to clipboard string generator

# Description
Added a copy button to the user's profile, which copies and generates the summary intro for managers, assistant managers and administrators. 
Copies the text to the user's clipboard, and checks if the user's in the team have submitted their weekly summaries, and categorizes based on that and saves it to the clipboard. 


<img width="371" alt="Screenshot 2023-10-03 at 7 34 02 PM" src="https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/22454953/8c84e40c-54f4-4376-bc62-8f5575f4c910">


## Main changes explained:
- Added a new button to the user's profile which generates the string to the clipboard. 
- Also, added a new permission seeSummaryIntro, to admins, assistant manager and managers. Volunteers cannot see the button (make sure to test it)
- The permission was added the admins and assistant manager on the backend, and works for owners as well. 
- Volunteers do not have the right by default.


## How to test:
1. check into current branch
2. do `npm install` and `...` to run this PR locally
3. log as admin/owner 
4. go to dashboard→ View profile 
5. Check if  "Generate Summary Intro" appears 
6. Click the button, and paste what appears. 
7. make sure your user is a part of a team with a manager. If no manager is assigned, the text will show "<Your Name>" instead 
8. Check if the user's who submitted a summary have and those who haven't are listed in the correct spot. 
9. Retest using a volunteer account and the button should not appear.

## Screenshots or videos of changes:
https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/22454953/d5bd3ff8-4916-4dd1-8e69-a6a217dfa218



## Note:
Make sure the admins, assistant manager have the right via an owner account, otherwise the button will not appear. 
<img width="1236" alt="Screenshot 2023-10-12 at 9 11 55 PM" src="https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/22454953/f5d08917-1f3e-482f-a18d-40a7494b6500">


They're also several errors and warnings please disregard them.
